vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java

```diff
@@ -500,8 +500,12 @@ public void run() {
                 if (last) {
                     terminateResponse();
                 }
-                callback.onComplete(VertxHttpExchange.this, context);
-                writeQueued = false;
+
+                if (callback != null) {
+                    callback.onComplete(VertxHttpExchange.this, context);
+                    writeQueued = false;
```

Comment:
This should always be set to false I think.

Reply:
Thanks Stuart, it is moved after the check.